### ISSUE TITLE
How to log during execute_bdd?

### ISSUE THREAD

**Issue by drmikecrowe** [AlexandreDecan/sismic#105]
Hey, I wanted to try logging from the listener during tests to a file.  I have my state machine `interpreter_klass`:

```py
        self.interpreter.attach(self.listener)

    def listener(self, e: MetaEvent):
        if e.name.startswith("step"):
            return
        logger.debug("StateMachine: " + repr(e))
```

The test log is filled with only the `parse` lines:

```
2021-03-01 07:09:44,914 [parse] DEBUG: format 'I do nothing' -> 'I do nothing'
2021-03-01 07:09:44,914 [parse] DEBUG: format 'I do nothing' -> 'I do nothing'
2021-03-01 07:09:44,915 [parse] DEBUG: format 'I reproduce "{scenario}"' -> 'I reproduce "(?P<scenario>.+?)"'
2021-03-01 07:09:44,916 [parse] DEBUG: format 'I reproduce "{scenario}"' -> 'I reproduce "(?P<scenario>.+?)"'
...
```

Does this make sense?  I'm positive my `logger.debug("StateMachine: " + repr(e))` is getting called, but it seems like it might be a different logger than the one I've configured in my test.

Is this possible?

**Comment by AlexandreDecan** [comment]
Hello,

Yes, this is possible. Behave (the underlying library we use for BDD) intercepts the loggers. Could you please try with the `no-logcapture` option as described here https://stackoverflow.com/questions/52781868/logging-in-behave-bdd-framework ?

Thanks!

**Comment by drmikecrowe** [comment]
Hmm, turns out this is tricky.  Here's some random findings:

First, I had to make the log setup part of a `Background:` scenario:

```py
@when(u"I setup logging")
def step_impl(context):
    my_setup_logging()
    context.config.setup_logging(
        level=logging.DEBUG,
        format="%(asctime)s %(filename)s[line:%(lineno)d] %(levelname)s %(message)s",
        datefmt="%Y-%m-%d %H:%M:%S",
    )
````
NOTE: You must call both the `context.config.setup_logging()` and your own logging initialization.

2. The behave params are not really needed:
```py
    return execute_bdd(
        statechart,
        [feature_dir],
        step_filepaths=steps,
        interpreter_klass=interpreter_with_class,
        # behave_parameters=["--no-logcapture", "--no-capture"],
    )
```

I thought there might be a way to set it up in my `interpreter_with_class`, but I can't seem to find the right combination.  The `interpreter` that is intiialized doesn't appear to have the same `conntext` reference that the `Background` step uses.

**Comment by AlexandreDecan** [comment]
Indeed, the `context` of a step is specific to Behave, while the `context` of an interpreter is specific to Sismic.
I've the impression that the issue came from Behave (notably because we do not mess with loggers in Sismic :-) 

One of the difficulty to configure Behave is related to how we integrated it into Sismic. Behave allows to define "hooks" that are automatically called (before scenarios, features, step, etc.) in `environment.py` and, unfortunately, we had to override this file to make Sismic and Behave work together. This implies that using bdd as provided by Sismic does not have all the "possibilities" offered by Behave, and makes a few things harder to do. 

However, I'm surprised how "complicated" it is to enable user-specific loggers in Behave....
